### PR TITLE
Clarify how to test in-progress SDK changes

### PR DESCRIPTION
**Pull request by jeremywiebe**
This change documents how to publish in-progress work to npmjs.org so that it can be tested with the scaffold in CircleCI.

(brief description / PR title)

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- (change1)

## Todos
- [ ] (if applicable) analytics events instrumented to measure impact of change

## How to test-drive this PR
- (necessary config changes)
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- (how to access the new / changed functionality -- fixtures, URLs)

### CODE REVIEW

**Comment by kevinxh** [comment]
Just curious how do we manage the X in `betaX` if we have multiple developers that try to publish a CircleCI test version of SDK.

For example: both develop A & B published `0.17.3 beta1`, will A's beta1 be overwritten?

**Comment by jeremywiebe** [comment]
@kevinxh Good question. With `npm` you cannot overwrite published packages so if you were following this and someone had already used the number you tried, they would encounter an error.

File: web/DEVELOPING.md [hunk 1]
```diff
@@ -20,7 +20,34 @@ npm link progressive-web-sdk
...
+       "progressive-web-sdk": "0.17.0@betaX"
+       ...
+   }
+   ```
+   
+  _**Note**: The version used in this step must match the version specified in step #2 and the tag used in step #3!_
+  
+5. Commit the change and push so that CircleCI runs a build. CircleCI will install the "beta" version of the progressive-web-sdk npm module which is your in-progress work.
+
+_**Note**: You will receive an error when publishing the beta package if another developer is also testing SDK changes with the scaffold and has used the number you chose for the betaX. If that happens you can either just try the next number (so if you tried @beta1 and it failed, try @beta2) or you can jump on Slack and ask what the last-numbered @beta is.
```
Comment (benlast) [inline web/DEVELOPING.md]: Do you also get an error if you try to merge to `develop` or `master` and you forgot to revert the `package.json` change and left the `betaX` in?

File: web/DEVELOPING.md [hunk 2]
```diff
@@ -20,7 +20,34 @@ npm link progressive-web-sdk
...
+
+This will leave these "beta" npm modules sitting around in the npm registry but that is probably safe as you can't use it unless you specifically install it by tag name. We could investigate `npm unpublish` if we're concerned about that. 
+  
 ## Swapping Integration Managers
 
 Note: This should make it's way into the Tutorial eventually, but this is here for now.
```
Comment (benlast) [inline web/DEVELOPING.md]: Line 53: _it's_ should be _its_

File: web/DEVELOPING.md (hunk 1 above)
Comment (benlast) [inline web/DEVELOPING.md]: Is there a convenient way to check which `betaX` values have been used? And do they have to be unique to the SDK version, or globally unique?

Comment (jeremywiebe) [inline web/DEVELOPING.md]: Noted in updated doc.  `npm view progressive-web-sdk versions`

Comment (jeremywiebe) [inline web/DEVELOPING.md]: For the SDK the release process will do the right thing even with a pre-release version in place. For the scaffold, no.  Might be a good thing for `circle.yml` to check.

**Pull request merged**